docs/v3/core/index.rst

```diff
@@ -1598,46 +1466,232 @@ Storage transformers may be stacked to combine different functionalities:
+strongly encouraged.
+
+Extension example
+-----------------
+
+The following example of array metadata demonstrates these extension naming schemes::
+
+    {
+        "zarr_format": 3,
+        "data_type": "string", // raw name, short-hand name
+        "chunk_key_encoding": {
+            "name": "default", // core
+            "configuration": { "separator": "." }
+        },
+        "codecs": [
+            {
+                "name": "vlen-utf8" // raw name
+            },
+            {
+                "name": "zstd", // raw name
+                "configuration": { ... }
+            }
+        ],
+        "chunk_grid": {
+            "name": "regular", // core
+            "configuration": { "chunk_shape": [ 32 ] }
+        },
+        "shape": [ 128 ],
+        "dimension_names": [ "x" ],
+        "attributes": { ... }
+    }
+
+Extension specifications
+------------------------
+
+Extensions SHOULD have a published specification. A published specification
+facilitates multiple implementations of an extension.
+
+For extensions with raw names, the `zarr-extensions`_ repository
+SHOULD either contain the specification or link to it.
+
+For extensions with URL-based names, it is RECOMMENDED that the URL resolve to
+a specification of the extension. Additionally, URL-based extensions MAY also register
+themselves under the `zarr-extensions`_ repository for better discovery.
+
+Because the ``fill_value`` metadata key is dependent on the data type, 
+extension data types SHOULD specify permitted values for the ``fill_value`` in
+their specification.
+
+
+.. _extension-guidance:
+
+Guidance for extension authors
+------------------------------
+
+*This section is non-normative and provides assistance for the authors of
+extensions, especially those who are just getting started.*
+
+* If you are just getting started, use the URL of your work-in-progress as an
+  identifier for your extension. The GitHub link, including the branch if you
+  would like, makes a fine choice. This says to the community that this is a
+  draft, and if they are interested in the details, they can follow the URL to
+  find out more.
+
+* When developing an extension for which you intend to register a short name,
+  you may wish to test it using the short name even before you have registered
+  it. However, you MUST register the name before using the extension for
+  non-test purposes/for purposes where interoperability with other
+  implementations/users is a concern.
+
+* If you are implementing a well-known extension like a data type or codec that
+  is already referred to by name in the community, you may want to check the `zarr-extensions`_
+  repository to see if someone has already implemented the extension.
+
+* For raw names that are coming from well-known projects, use the same prefix followed
+  by a dot for requesting your raw name, e.g. "numcodecs.". Other examples of prefixes can
+  be found in the `zarr-extensions`_ repository.
```

Comment:
Note that these would be considered namespaced extensions under my proposal.

Reply:
related to my [comment above](https://github.com/zarr-developers/zarr-specs/pull/330#discussion_r1983527438) but we should clarify early on that registered names can also be namespaced.